spec/lib/ruby_smb/client_spec.rb

```diff
@@ -223,15 +225,11 @@
 
         it 'calls #smb2_sign if it is an SMB2 client' do
           allow(smb2_client).to receive(:is_status_pending?).and_return(false)
+          allow(smb2_client).to receive(:signing_required).and_return(true)
+          allow(smb2_client).to receive(:session_key).and_return(Random.new.bytes(16))
           expect(smb2_client).to receive(:smb2_sign).with(smb2_request).and_call_original
           smb2_client.send_recv(smb2_request)
         end
-
-        it 'calls #smb3_sign if it is an SMB3 client' do
-          allow(smb3_client).to receive(:is_status_pending?).and_return(false)
-          expect(smb3_client).to receive(:smb3_sign).with(smb2_request).and_call_original
-          smb3_client.send_recv(smb2_request)
-        end
```

Comment:
Are we testing sm3_sign elsewhere?

Reply:
That broke because the original `smb3_sign` function would check to see if the packet should be signed or not and only sign it when appropriate. Since I moved that into a mixin so it could be used by both, I switched the logic to **always** sign and made it the caller's responsibility to determine whether or not the signing operation should be performed. This was necessary because the check is different for the client and server. I also thought it was a little misleading that the function named `smb3_sign` may or may not actually sign the data 🤷🏻 .

Commit for context: 8998b4856fab1ea009282cb1f0d8eb204e58ce08

FWIW, the original test was calling the `smb3_sign` function, but the signing wasn't actually being performed so I think the value it was providing was negligible.

Reply:
Agreed, especially with the `smb3_sign` only sometimes actually signing